### ISSUE TITLE
Add support to Forward Gradients

### ISSUE THREAD

**Issue by morgoth95** [nebuly-ai/nebulgym#3]
Add support to the forward gradient computation technique. This technique allows gradients to be estimated during the forward pass without the need for an explicit backward calculation. More information is available in the original [paper](https://arxiv.org/pdf/2202.08587.pdf). An open-source implementation can be found [here](https://github.com/orobix/fwdgrad).

**Comment by berkecanrizai** [comment]
I am not sure if it's any help but, it is already implemented in JAX. See: [https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#jacobian-vector-products-jvps-aka-forward-mode-autodiff](Jacobian-Vector products (JVPs, aka forward-mode autodiff)).
